components/report-manager-service/worker/worker.go

```diff
@@ -249,6 +250,53 @@ func (t *GenerateReportTask) Run(ctx context.Context, task cereal.Task) (interfa
 		}
 	}
 
+	//Getting the bucket lifecycle configurations
+	lifecycles, err := t.ObjStoreClient.GetBucketLifecycle(ctx, job.RequestToProcess.RequestorId)
+
+	if lifecycles == nil && err.Error() == "The lifecycle configuration does not exist" {
+		config := lifecycle.NewConfiguration()
+		config.Rules = []lifecycle.Rule{
+			{
+				ID:     "expire-bucket",
+				Status: "Enabled",
+				Expiration: lifecycle.Expiration{
+					Days: 1,
+				},
+			},
+		}
+		//Setting the lifecycle configurations
+		err = t.ObjStoreClient.SetBucketLifecycle(ctx, job.RequestToProcess.RequestorId, config)
+		if err != nil {
+			return nil, err
+		}
+	} else if err != nil {
+		return nil, err
+	}
+
+	if lifecycles != nil {
+		flag := false
+		for i := 0; i < len(lifecycles.Rules); i++ {
+			if !lifecycles.Rules[i].Expiration.IsNull() {
```

Comment:
The condition should check for Date and Days as well.
 some thing like Expiration.IsDateNull() || Expiration.IsDaysNull()